src/engine/x86-64/X86_64Stack.v3

```diff
@@ -151,32 +152,39 @@ class X86_64Stack extends WasmStack {
 	def setRsp(ptr: Pointer) -> this {
 		rsp = ptr;
 	}
-	// This function returns true if a handler IS NOT found, and otherwise
-	// returns {(stp_idx, order)}.
-	def findSuspendHandler(frame: TargetFrame, tag: Tag) -> bool {
+	def findSuspensionHandler(kind: SuspensionType, tag: Tag) -> ExHandler {
```

Comment:
Can we split this into two steps? One issue is that this method has the same name as a subroutine used here, but this one also performs a side effect in case of suspension.